crypto/data_formats.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crypto
+
+import (
+	"encoding/base64"
+	"strconv"
+
+	"github.com/benlaurie/objecthash/go/objecthash"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/sigpb"
+)
+
+// This file contains struct specific mappings and data structures.
+// TODO: remove data-structure specific operations.
+
+// Constants used as map keys when building input for ObjectHash. They must not be changed
+// as this will change the output of hashRoot()
+const (
+	mapKeyRootHash       string = "RootHash"
+	mapKeyTimestampNanos string = "TimestampNanos"
+	mapKeyTreeSize       string = "TreeSize"
+)
+
+// HashTrillianSignedLogRoot hashes SignedLogRoot objects in a custom way.
+func HashTrillianSignedLogRoot(root trillian.SignedLogRoot) []byte {
```

Comment:
I added the TODO and assigned it to @AlCutter.
It's my opinion that we should remove everything in this file and make signing a generic thing rather than creating new functions for every object we want to sign.  But that should be a separate PR.  Just separating it here so it's easier to remove.